File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> s, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    int first = 0;
    int second = 0;

    for(int i=0; i<s.size(); i++)
        pq.push(s[i]);
    
    while(pq.size()>1 && pq.top()<K){
        first = pq.top();
        pq.pop();
        
        second = pq.top();
        pq.pop();
        
        pq.push(first+(second * 2));
        answer++;
    }
    
    if(pq.top()<K)
        answer = -1;
    
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] s, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int num : s){
            pq.offer(num);
        }
        
        while(pq.size() > 1 && pq.peek() < K){
            int first = pq.poll();
            int second = pq.poll();
            
            int mixed = first + (second * 2);
            pq.offer(mixed);
            answer++;
        }
        
        if(pq.peek() < K)
            return -1;
        return answer;
    }
}